### ISSUE TITLE
sam build Maven project uses the wrong java version

### ISSUE THREAD

**Issue by jaxley** [aws/aws-toolkit-jetbrains#1803]
**Describe the bug**

Running an AWS java lambda from within IntelliJ fails at the build step because when sam checks /usr/local/bin/mvn, it reports a different jdk version than is configured on the project, in the pom, and in the JAVA_HOME environment variable (set on the Run configuration)

Should run the build with Java 1.8, but is trying to use Java 13.

```
Found workflow 'JavaMavenWorkflow' to support capabilities 'Capability(language='java', dependency_manager='maven', application_framework=None)'
/usr/local/bin/mvn is using a JVM with major version 13 which is newer than 8 that is supported by AWS Lambda. The compiled function code may not run in AWS Lambda unless the project has been configured to be compatible with Java 8 using 'maven.compiler.target' in Maven.
Running
```

```
[ERROR] Failed to execute goal on project my-app: Could not resolve dependencies for project my-appjar:1.0-SNAPSHOT: Could not find artifact com.sun:tools:jar:1.8 at specified path /usr/local/Cellar/openjdk/13.0.2+8_2/libexec/openjdk.jdk/Contents/Home/../lib/tools.jar -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
```

**To reproduce**
1. Run intellij from launcher. Default JDK on system is 11
```
$ /usr/libexec/java_home
/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
```
2. Install AWS toolkit and sam per instructions
3. Create run configuration for a Java 1.8 Lambda application
3a. Java application has all versions in the pom.xml set already for 1.8.
4. Run the Run or Debug configuration. It will start the sam build, that will generate the above errors

```
$ cat template.yaml 
AWSTemplateFormatVersion : '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Description: A sample SAM template for deploying Lambda functions.

Parameters:
  CodeUri:
    Description: CodeUri - change this to pom.xml directory for AWS Toolkit IDE plugin with --parameter-overrides "CodeUri"
    Type: String
    Default: lambda/target/foo.jar

Resources:
  LambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: !Sub ${CodeUri}
      Handler: foo::handleRequest
      Runtime: java8
      MemorySize: 512
      Timeout: 900
#      AutoPublishAlias: live
# Grants this function permission to call lambda:InvokeFunction
      Policies:
        - Version: "2012-10-17"
          Statement: 
          - Effect: "Allow"
            Action: 
              - "lambda:InvokeFunction"
            Resource: '*'
```

**Expected behavior**
I'd expect to have the ability to ensure that JAVA_HOME is set to a consistent value for the application.  This could be done in many possible ways:
1. Defer to the version in the pom.xml
2. Defer to the version set on the Intellij project
3. Defer to the JAVA_HOME environment variable that is set in the AWS Toolkit Run configuration screen
4. Allow an option to explicitly set the SDK you want to use.

**Screenshots**
![image](https://user-images.githubusercontent.com/414771/83192601-a046b000-a0ea-11ea-8242-b841299d175c.png)
![image](https://user-images.githubusercontent.com/414771/83192625-a76dbe00-a0ea-11ea-8236-635462a393ae.png)

**Your Environment**
- OS: Catalina 10.15.4 (19E287)
- JetBrains product: IntelliJ IDEA
- JetBrains product version:  2020.1.1
- AWS Toolkit version: 1.16 (2020-05-27)
- SAM CLI version: 0.52.0
- JVM/Python version: 1.8.0_242
- Maven version:  Apache Maven 3.6.3

**Comment by gitjhb** [comment]
yeah I got the same issue

**Comment by ckant** [comment]
Source of the log message
https://github.com/awslabs/aws-lambda-builders/blob/24f0aca6ed41ec2b08333655adbd1433f14cdc9e/aws_lambda_builders/workflows/java_maven/maven_validator.py#L39

**Comment by jaxley** [comment]
Yes. That gets imported from the pom settings as well, which are set
properly

**Comment by abrooksv** [comment]
Created a PR to take the Project's SDK and pass it to SAM CLI as the `JAVA_HOME` environment variable. Confirmed that it used JDK 8 even when my default is 11

**Comment by abrooksv** [comment]
This has been released in 1.17
